src/components/organisms/AssetActions/Consume.tsx

```diff
@@ -104,12 +111,17 @@ export default function Consume({
       ) : (
         <>
           <Button style="primary" onClick={handleConsume} disabled={isDisabled}>
-            {hasDatatoken || hasPreviousOrder ? 'Download' : 'Buy'}
+            {hasPreviousOrder
```

Comment:
now that we have this slightly revised logic, next step is to incorporate the timeout here right? Something with comparing time of previous order with the timeout in DDO?

Reply:
the timeout logic is already handled in `checkPreviousOrder`. So if timeout is expired ( you understand what i mean ), `checkPreviousOrder` will not return anything so you `hasPreviousOrder` will be `false`

Reply:
oooh, amazing info! Considered all that and tweaked the help text logic instead